src/gui/properties/peerlistwidget.cpp

```diff
@@ -57,6 +57,19 @@
 #include "propertieswidget.h"
 #include "uithememanager.h"
 
+namespace BitTorrent
+{
+    bool operator==(const BitTorrent::PeerAddress &left, const BitTorrent::PeerAddress &right)
+    {
+        return (left.ip == right.ip) && (left.port == right.port);
+    }
+
+    uint qHash(const BitTorrent::PeerAddress &addr, const uint seed)
+    {
+        return qHash(addr.ip, seed) ^ addr.port;
```

Comment:
I think the following would be better:
```c++
const QString endpoint = addr.ip.toString() + ':' + QString::number(addr.port);
return qHash(endpoint, seed) 
```

Reply:
I picked the 2nd but i had to use `::qHash` otherwise it throws some error.